core/watcher/eventqueue/notify.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package eventqueue
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/juju/errors"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/core/changestream"
+)
+
+// NotifyWatcher watches for changes to a database table.
+type NotifyWatcher struct {
+	*BaseWatcher
+
+	out       chan struct{}
+	tableName string
+	selectAll string
+}
+
+// NewNotifyWatcher returns a new watcher that receives changes ...
+func NewNotifyWatcher(base *BaseWatcher, tableName, scope string) *NotifyWatcher {
+	w := &NotifyWatcher{
+		BaseWatcher: base,
+		out:         make(chan struct{}),
+		tableName:   tableName,
+		selectAll:   fmt.Sprintf("SELECT %s FROM %s", scope, tableName),
```

Comment:
can we make this a constant somewhere? I guess we'll need it elsewhere probably, instead of copying all over

Reply:
Officially, we require it to be placed here. This is on purpose. The idea being it's close to the call site.